### PR TITLE
Add using account data in transfer hook example

### PR DESCRIPTION
**Pull request by Woody4618**
### Problem
There are many people asking how to derive a token account owner in a transfer hook.

### Solution
This guide section explains this and there is also a new example showing how to do this.

### CODE REVIEW

File: content/guides/token-extensions/transfer-hook.md [hunk 1]
```diff
@@ -1549,6 +1549,146 @@ Running tests...
...
+
+Sometimes you may want to use account data to derive additional accounts in the
+extra account metas. This is useful if, for example, you want to use the token
+account's owner as a seed for a PDA.
+
+When creating the ExtraAccountMeta you can use the data of any account as an
+extra seed. In this case we want to derive a counter account from the token
+account owner and the string 'counter'.
+
+This is how you set it up in the InitializeExtraAccountMetaList trait.
```
Comment (ZYJLiu) [inline content/guides/token-extensions/transfer-hook.md:1562]:
```suggestion
This is how you set it up in the `extra_account_metas()` function.
```

InitializeExtraAccountMetaList is a struct (not a trait)

File: content/guides/token-extensions/transfer-hook.md [hunk 2]
```diff
@@ -1549,6 +1549,146 @@ Running tests...
...
+            ]
+        )
+    }
+}
+```
+
+Let's look at the token account struct to understand how the account data is
+stored. Below is an example of a token account structure. So we can take 32
+bytes at position 32 to 64 as the owner of the token account, which is at
+'account_index: 0'.
```
Comment (ZYJLiu) [inline content/guides/token-extensions/transfer-hook.md:1589]: Could you clarify what you mean by "which is at
'account_index: 0'"?

We are getting the address of the `owner`, which is bytes 32-64 in the Account struct, but not sure what 'account_index: 0' means.

File: content/guides/token-extensions/transfer-hook.md [hunk 3]
```diff
@@ -1549,6 +1549,146 @@ Running tests...
...
+    pub delegate: COption<Pubkey>,
+    pub state: AccountState,
+    pub is_native: COption<u64>,
+    pub delegated_amount: u64,
+    pub close_authority: COption<Pubkey>,
+}
+```
+
+Now we need to put the extra account metas in the InitializeExtraAccountMetaList
+struct.
```
Comment (ZYJLiu) [inline content/guides/token-extensions/transfer-hook.md:1611]: Can you clarify what you mean? Is this referring to using `extra_account_metas()` to calculate the space for the new `extra_account_meta_list` account or something else?

File: content/guides/token-extensions/transfer-hook.md [hunk 4]
```diff
@@ -1549,6 +1549,146 @@ Running tests...
...
+    pub mint: InterfaceAccount<'info, Mint>,
+    #[account(init, seeds = [b"counter", payer.key().as_ref()], bump, payer = payer, space = 16)]
+    pub counter_account: Account<'info, CounterAccount>,
+    pub token_program: Program<'info, Token2022>,
+    pub associated_token_program: Program<'info, AssociatedToken>,
+    pub system_program: Program<'info, System>,
+}
+```
+
+and then to the remaining accounts of the transfer hook accounts.
```
Comment (ZYJLiu) [inline content/guides/token-extensions/transfer-hook.md:1639]:
```suggestion
And then to the remaining accounts of the transfer hook accounts.
```

What does remaining accounts mean in this context? Use "remaining accounts" may be confused with unchecked accounts accessible from `ctx.remaining_accounts`

File: content/guides/token-extensions/transfer-hook.md [hunk 5]
```diff
@@ -1549,6 +1549,146 @@ Running tests...
...
+    pub owner: UncheckedAccount<'info>,
+    /// CHECK: ExtraAccountMetaList Account,
+    #[account(seeds = [b"extra-account-metas", mint.key().as_ref()], bump)]
+    pub extra_account_meta_list: UncheckedAccount<'info>,
+    #[account(seeds = [b"counter", owner.key().as_ref()], bump)]
+    pub counter_account: Account<'info, CounterAccount>,
+}
+```
+
+In the client this account is auto generated and you can use it as follows.
```
Comment (ZYJLiu) [inline content/guides/token-extensions/transfer-hook.md:1659]: Which account is "this account"?

File: content/guides/token-extensions/transfer-hook.md [hunk 6]
```diff
@@ -1549,6 +1549,146 @@ Running tests...
...
+```js
+const [counterPDA] = PublicKey.findProgramAddressSync(
+  [Buffer.from("counter"), wallet.publicKey.toBuffer()],
+  program.programId,
+);
+```
+
+The counter PDA in this case is created in the initialize hook. If you want this
+to work for every account you probably need to add a function on your website to
+create this account beforehand.
```
Comment (ZYJLiu) [inline content/guides/token-extensions/transfer-hook.md:1690]: Can you clarify what you mean by "If you want this to work for every account"? What is "this" referring to?

**Comment by Woody4618** [comment]
I tried to explain it a bit better. Let me know if it makes more sense now please

**Pull request merged**